crates/turbopack-ecmascript-runtime/js/src/shared/runtime-utils.ts

```diff
@@ -168,11 +181,37 @@ function interopEsm(
   esm(ns, getters);
 }
 
-function esmImport(sourceModule: Module, id: ModuleId): EsmNamespaceObject {
+function esmImport(
+  sourceModule: Module,
+  id: ModuleId
+): EsmNamespaceObject | (Promise<EsmNamespaceObject> & AsyncModuleExt) {
   const module = getOrInstantiateModuleFromParent(id, sourceModule);
   if (module.error) throw module.error;
   if (module.namespaceObject) return module.namespaceObject;
   const raw = module.exports;
+
+  if (isPromise(raw)) {
+    const promise = raw.then((e) => {
+      const ns = {};
+      interopEsm(e, ns, e.__esModule);
+      return ns;
+    });
+
+    module.namespaceObject = Object.assign(promise, {
+      get [turbopackExports]() {
+        return raw[turbopackExports];
+      },
+      get [turbopackQueues]() {
+        return raw[turbopackQueues];
+      },
+      get [turbopackError]() {
+        return raw[turbopackError];
+      },
+    } satisfies AsyncModuleExt);
+
+    return module.namespaceObject;
+  }
```

Comment:
A commonjs module exporting a Promise under `module.export` should not alter the Promise in any way.
Only Promises returned by ESM should be altered.

See interop test: https://sokra.github.io/interop-test/#single-promise-object-with-default-export
All bundlers and node agree on that here.

Reply:
That should work now, I wonder how webpack handles that when you also have async modules